private/pkg/connect/config.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	connect_go "github.com/bufbuild/connect-go"
+)
+
+type ClientConfig struct {
+	httpClient              connect_go.HTTPClient
+	addressMapper           func(string) string
+	interceptors            []connect_go.Interceptor
+	authInterceptorProvider func(string) connect_go.UnaryInterceptorFunc
+}
+
+func NewClientConfig(httpClient connect_go.HTTPClient, options ...ClientConfigOption) *ClientConfig {
+	var cfg ClientConfig
+	for _, opt := range options {
+		opt(&cfg)
+	}
+	return &cfg
+}
```

Comment:
```suggestion
func NewClientConfig(httpClient connect_go.HTTPClient, options ...ClientConfigOption) *ClientConfig {
	cfg  := ClientConfig{
		httpClient: httpClient,
	}
	for _, opt := range options {
		opt(&cfg)
	}
	return &cfg
}
```

Reply:
httpClient appears to be ignored and then used in `MakeClient` later.

Reply:
🤦, good catch.

Indeed, nothing in this repo's tests actually exercises remote generation in a way that would have caught this. But I just tried it out and it panics.